insights/parsers/nfs_exports.py

```diff
@@ -189,19 +141,34 @@ def __iter__(self):
     @staticmethod
     def reconstitute(path, d):
         """
-        Reconstruct a line from its parsed value.  Original whitespace is not
-        maintained; instead two spaces (``"  "``) is used between each word.
+        'Reconstitute' a line from its parsed value.  The original lines are
+        not used for this.  The hosts in d are listed in alphabetical order,
+        and the options are listed in the order originally given.
+
+        This function is deprecated.  Please use the `raw_lines` dictionary
+        property of the parser instance instead, as this contains the actual
+        lines from the exports file.
 
-        Args:
-            path (str): The export path
-            d (dict): The dictionary that's the value of the exported path key
-                      in the parsed dict.
+        Arguments:
+            path (str): The exported path
+            d (dict): The hosts definition of the exported path
 
         Returns:
-            str: The reconstituted line from the exports file
+            str: A line simulating the definition of that exported path to
+            those hosts.
         """
-        return "  ".join([path] + ["%s(%s)" % (host, ",".join(options))
-                         for host, options in d.iteritems()])
+        deprecated(
+            NFSExportsBase.reconstitute,
+            'Please use the `raw_lines` dictionary property of the parser instance'
+        )
+        return "{path}  {hosts}".format(
+            path=path,
+            hosts=' '.join((
+                '{host}({opts})'.format(
+                    host=h, opts=','.join(d[h]))
+                for h in sorted(d)
+            ))
+        )
```

Comment:
@PaulWay this change to the return creates a different result than the original code, would you mind just leaving the original return code in this method.

Reply:
Hi @bfahr - the problem with the old code is that it's difficult to test.  Because it uses `iteritems()`, the order in which the hosts are listed is random and may vary.  It's not very pretty but it's tested now.  Let me know what you think of this update.